core/shortcut_items.js

```diff
@@ -94,15 +94,15 @@ const registerCopy = function() {
     name: names.COPY,
     preconditionFn: function(workspace) {
       return !workspace.options.readOnly && !Gesture.inProgress() &&
-          Blockly.selected && Blockly.selected.isDeletable() &&
-          Blockly.selected.isMovable();
+          common.getSelected() && common.getSelected().isDeletable() &&
+          common.getSelected().isMovable();
```

Comment:
I wonder if this is the point at which, even if we can't use optional chaining, we ought instead to do `const selected = common.getSelected();` and then just check `selected.isDeletable() && selected.isMovable()`.

Reply:
Done